components/layout/layout.tsx

```diff
@@ -48,10 +45,28 @@ function generator({ suffixCls, tagName, displayName }: GeneratorProps) {
 }
 
 const Basic = React.forwardRef<HTMLElement, BasicPropsWithTagName>((props, ref) => {
-  const { prefixCls, className, children, tagName, ...others } = props;
-  const classString = classNames(prefixCls, className);
+  const {
+    prefixCls: customizePrefixCls,
+    suffixCls,
+    className,
+    tagName: TagName,
+    ...others
+  } = props;
+
+  const { getPrefixCls } = React.useContext(ConfigContext);
+  const prefixCls = getPrefixCls('layout', customizePrefixCls);
+
+  const [wrapSSR, hashId] = useStyle(prefixCls as string);
+
+  const prefixWithSuffixCls = suffixCls ? `${prefixCls}-${suffixCls}` : prefixCls;
 
-  return React.createElement(tagName, { className: classString, ...others, ref }, children);
+  return wrapSSR(
+    <TagName
+      className={classNames(customizePrefixCls || prefixWithSuffixCls, className, hashId)}
```

Comment:
这里是不是用`??`好一点，是不是会有外界传过来的`prefixCls`为`''`的情况？
```suggestion
      className={classNames(customizePrefixCls ?? prefixWithSuffixCls, className, hashId)}
```

Reply:
不会的，`prefixCls` 本身就是内部属性。传入 `''`那都没办法魔改样式了。

Reply:
ok